content/document.js

```diff
@@ -197,32 +198,70 @@ function unarchive(document, move) {
   return created;
 }
 
-const read = memoize((folder) => {
+const read = memoize((folderOrFilePath) => {
```

Comment:
Wondering if this PR should not be limited to changing `findAll()`? From what I understand this changes `read()`'s API, don't we have callers that are passing folders in?

Reply:
Ah I see, I was only looking at the if-branch. Still wondering though whether we can reap the findAll-speed improvements without changing fn

Reply:
You're absolutely right! 
But... may I ask to make an exception to that general rule here? Pleeeease. 

The argument is that the two most important uses of this code is when we do prod builds which is a big `yarn build`, and when we do PR CI which builds the specified files (based on the git diff). 
In both those cases it's actually smart to pass in a full absolute file path to `read()` because you don't need to loop through the roots to figure out where it came from. 
But yes, what could happen is that the `yarn build` causes a call of `read('/home/path/to/absolute/content/files/en-us/foo/index.html')` first and then later some KS sidebar or some other code internally triggers a call to `read('/en-us/foo')` which is the same result but a different memoization key. But I would argue that the cache is not important enough to protect against that. The `read()` is so fast and because of the LRU (max 2,000 entries) you'll end up calling the `read()` with the same key many times anyway and not benefitting from the memoization. 

So I violated a bit by changing the `findAll()` *and* the `read()` and that's not great. (But hopefully my explanation and justification can make up for that). 
However, perhaps the correct API is that `read()` should always and only expect a `filePath` as the one and only argument. And if you only have a folder (e.g. `fr/web/html/element/video`) you should be using some other `readByFolder()` wrapper that can turn `fr/web/html/element/video` into `/full/path/to/translated-content/files/fr/web/html/element/video/index.html` before it goes into `read()`.

Reply:
Oh I didn't even consider the bits about the cache. Maybe we should then as part of this factor out the finding of the correct path so that the cache still gets hit? Or do you prefer getting this in first? It seems worthwhile to me to reduce the responsibilities of `read` a little